src/main/java/org/opentripplanner/street/search/state/State.java

```diff
@@ -52,6 +53,12 @@ public class State implements AStarState<State, Edge, Vertex>, Cloneable {
   // we should DEFINITELY rename this variable and the associated methods.
   public double walkDistance;
 
+  // how far a sharing vehicle powered by battery has driven
+  public double drivenBatteryMeters;
+
+  //the available battery distance of a currently selected sharing vehicle
+  public Optional<Double> currentRangeMeters;
```

Comment:
For reasons that I don't quite understand it's totally unusual in Java to have instance fields that are Optionals. You have to use a regular double with a magic value for "no value".

Reply:
It's probably also better for performance.

Reply:
hmm, would it be ok, to use Double(using null as "no value"), the wrapper class instead of the primitive double?
Because using a magic value for "no value" sounds slightly ugly.

Reply:
I know that it's ugly but we do that quite a bit in OTP to get the absolute best performance. This is a very hot code path.

Reply:
> Also please make as many fields as possible private.

sounds good to me. We weren't sure here because the other fields in this class were public. But if you don't see a problem here, we'll change the added field.

Reply:
> I know that it's ugly but we do that quite a bit in OTP to get the absolute best performance. This is a very hot code path.

Okay. What do you think would be a good value?
Zero would be really bad because if the battery was emty, we would ignore our logic and it would use that vehicle.
Would Double.MAX_VALUE or DOUBLE.MIN_VALUE be a good value? 
Or maybe a negative number, then we would have to change the check with the negative number.

Reply:
We made it a basic double and used Double.POSITIVE_INFINITY as the initializing value